### PR TITLE
Validate master/worker config

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Add validation to master worker config

## Brief change log

- Validate master worker config properties.
- Change all time properties in config to Duration.

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java [hunk 1]
```diff
@@ -17,174 +17,87 @@
...
-    public void setKillYarnJobWhenTaskFailover(boolean killYarnJobWhenTaskFailover) {
-        this.killYarnJobWhenTaskFailover = killYarnJobWhenTaskFailover;
+    private Duration stateWheelInterval = Duration.ofMillis(5);
+    private double maxCpuLoadAvg = -1;
+    private double reservedMemory = 0.3;
+    private Duration failoverInterval = Duration.ofMinutes(10);
+    private boolean killYarnJobWhenTaskFailover = true;
+
+    @PostConstruct
+    public void validate() {
```
Comment (kezhenxu94) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java]: Can you implement the validation logic via implementing `org.springframework.validation.Validator` interface and add annotation `org.springframework.validation.annotation.Validated` to this class? That's what Spring propose to validate properties

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java [hunk 2]
```diff
@@ -21,170 +21,108 @@
...
+            errors.rejectValue("exec-threads", null, "should be a positive value");
+        }
+        if (masterConfig.getDispatchTaskNumber() <= 0) {
+            errors.rejectValue("dispatch-task-number", null, "should be a positive value");
+        }
+        if (masterConfig.getHeartbeatInterval().toMillis() < 0) {
+            errors.rejectValue("heartbeat-interval", null, "should be a valid duration");
+        }
+        if (masterConfig.getTaskCommitRetryTimes() <= 0) {
+            errors.rejectValue("task-commit-retry-times", null, "should better a positive value");
```
Comment (kezhenxu94) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java]:
```suggestion
            errors.rejectValue("task-commit-retry-times", null, "should be a positive value");
```

**Review by kezhenxu94 — DISMISSED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java]: Done

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/config/MasterConfig.java]: Done

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**